providers/env-var/src/main/java/dev/openfeature/contrib/providers/envvar/OS.java

```diff
@@ -0,0 +1,11 @@
+package dev.openfeature.contrib.providers.envvar;
+
+/**
+ * This internal class abstracts away Java's {@link System} class for test purposes.
+ * It is not intended to be used directly.
+ */
+class OS {
```

Comment:
Minor comment (potential improvement)

We could make this an interface (contract) and have a default implementation based on `System.getenv(name)`

That would allow having custom env variable handling (ex:- prefixes, casing and even sources like spring configurations which come from different sources [1] ) and still use the same provider 

Could be a follow up work :) 

[1] - https://docs.spring.io/spring-boot/docs/1.0.1.RELEASE/reference/html/boot-features-external-config.html